ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -2399,12 +2399,16 @@ private void getMetaData(QB qb, ReadEntity parentInput)
         }
 
         boolean isTableWrittenTo = qb.getParseInfo().isInsertIntoTable(ts.tableHandle.getDbName(),
-            ts.tableHandle.getTableName(), ts.tableHandle.getBranchName());
+            ts.tableHandle.getTableName(), ts.tableHandle.getSnapshotRef());
         isTableWrittenTo |= (qb.getParseInfo().getInsertOverwriteTables().
             get(getUnescapedName((ASTNode) ast.getChild(0), ts.tableHandle.getDbName()).toLowerCase()) != null);
         assert isTableWrittenTo :
             "Inconsistent data structure detected: we are writing to " + ts.tableHandle  + " in " +
                 name + " but it's not in isInsertIntoTable() or getInsertOverwriteTables()";
+        Boolean isTableTag = Optional.ofNullable(ts.tableHandle.getSnapshotRef()).map(HiveUtils::checkTableTag)
+            .orElse(false);
+        assert isTableWrittenTo && !isTableTag : "Don't support write (insert/delete/update/merge) to iceberg tag " +
```

Comment:
assert would be removed in runtime, show we throw UnsupportedOperatiionEException here?

Reply:
Fixed.